resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -786,6 +790,15 @@ private[spark] class MesosClusterScheduler(
     }
   }
 
+  /**
+   * Check if the task is outdated i.e. has already been launched or is pending
+   * If neither, the taskId is outdated and should be ignored
+   * This is to avoid scenarios where an outdated status update arrives
+   * after a supervised driver has already been relaunched
+   */
+  private def isTaskOutdated(taskId: String, state: MesosClusterSubmissionState): Boolean =
+    taskId != state.taskId.getValue && !pendingRetryDrivers.contains(state.driverDescription)
```

Comment:
@samvantran . This works, but `pendingRetryDrivers` are assumed to be keyed by submission id. We should use `pendingRetryDrivers.exists(_.submissionId == state.driverDescription.submissionId)` instead of `pendingRetryDrivers.contains` to be robust. Could you fix like that?

Reply:
Updated!